packages/runtime/src/next/client.ts

```diff
@@ -658,45 +674,41 @@ export class Makeswift {
   async getComponentSnapshot(
     id: string,
     {
-      siteVersion = MakeswiftSiteVersion.Working,
+      siteVersion: siteVersionPromise,
       locale,
-    }: {
-      siteVersion?: MakeswiftSiteVersion
-      locale?: string
-    } = {},
+    }: { siteVersion: MakeswiftSiteVersion | Promise<MakeswiftSiteVersion>; locale?: string},
   ): Promise<MakeswiftComponentSnapshot> {
     const searchParams = new URLSearchParams()
     if (locale) searchParams.set('locale', locale)
 
+    const siteVersion = await siteVersionPromise
     const response = await this.fetch(
-      `v1/element-trees/${id}?${searchParams.toString()}`,
+      `v2/element-trees/${id}?${searchParams.toString()}`,
       siteVersion,
     )
 
     // If the element tree is not found, we generate a document with null data
     if (response.status === 404) {
-      const apiKeyPrefix = z.string().parse(this.apiKey.split('-').at(0))
-
-      return {
-        document: {
-          id,
-          locale: locale ?? null,
-          data: null,
-          // We pass the seed to generate the elementKey to make sure it's unique for each site.
-          seed: apiKeyPrefix,
-        },
-        cacheData: CacheData.empty(),
-      }
+      return this.getDummyDocument(id, locale)
     }
 
     const json = await response.json()
 
     if (!response.ok) {
-      console.error('Failed to get page snapshot', json)
-      throw new Error(`Failed to get page snapshot with error: "${response.statusText}"`)
+      console.error('Failed to get component snapshot', json)
+      throw new Error(`Failed to get component snapshot with error: "${response.statusText}"`)
+    }
+
+    const elementTreeResponse = getMakeswiftComponentDocumentSchemaV2.parse(json)
+
+    // favor picking 'elementTree' if it is available.
+    const document = elementTreeResponse.elementTree ?? elementTreeResponse.parentLocaleElementTree
+
+    // This case should never be reached (response was an http 200, but neither the element tree nor the parent locale element tree were provided in the response)
```

Comment:
If this case should never be reached, we should model that in our parsing such as to have the type system enforce it. I think this is better than having a comment + fallback code

Reply:
Added to the zod type for this